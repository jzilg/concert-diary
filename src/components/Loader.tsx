import React, { ReactElement } from 'react'
import style from './loader.css'

function Loader(): ReactElement {
    return <div className={style.bar} />
}

export default Loader
