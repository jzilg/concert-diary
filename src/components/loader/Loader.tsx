import type { FC } from 'preact/compat'
import React from 'preact/compat'
import style from './loader.module.scss'

export type Props = {
    isLoading: boolean
}

const Loader: FC<Props> = (props) => {
    const { isLoading } = props

    if (!isLoading) {
        return null
    }

    return <div className={style.bar} />
}

export default Loader
