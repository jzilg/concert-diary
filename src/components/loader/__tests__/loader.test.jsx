import React from 'preact/compat'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Loader from '../Loader'

describe('Loading', () => {
    it('should render correctly if isLoading is true', () => {
        const component = shallow(<Loader isLoading />)
        const tree = shallowToJson(component)

        expect(tree).toMatchSnapshot()
    })

    it('should render correctly if isLoading is false', () => {
        const component = shallow(<Loader isLoading={false} />)
        const tree = shallowToJson(component)

        expect(tree).toMatchSnapshot()
    })
})
