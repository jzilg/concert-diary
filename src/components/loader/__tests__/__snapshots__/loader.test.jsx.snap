// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly if isLoading is false 1`] = `""`;

exports[`Loading should render correctly if isLoading is true 1`] = `
<div
  class="bar"
  className="bar"
/>
`;
